upload/upload_registrar.py

```diff
@@ -1,8 +1,141 @@
-    # TODO parse sample_registrar_data.txt and upload to db. 
+Args:
+    filename: registrar provided file of new students
+    database: select "dev" or "prod" database
+    -v: increase output verbosity
+
+Written by Michael Huynh
+July 05, 2017
+"""
+
+# import all that good stuff
+import argparse, re, datetime, db
+from sqlalchemy import Column, Integer, String, Date
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+Base = declarative_base()
+
+class Member(Base):
+    __tablename__ = "members";
+    
+    user_id = Column(Integer, primary_key = True)
+    uid = Column(String, unique = True)
+    last_name = Column(String)
+    first_name = Column(String)
+    preferred_name = Column(String)
+    middle_name = Column(String)
+    email = Column(String)
+    phone = Column(String)
+    gender = Column(Integer)
+    gender_custom = Column(String)
+    birthday = Column(Date)
+    entry_year = Column(Integer)
+    graduation_year = Column(Integer)
+    msc = Column(Integer)
+    building = Column(String)
+    room_num = Column(Integer)
+    address = Column(String)
+    city = Column(String)
+    state = Column(String)
+    zip = Column(String)
+    country = Column(String)
+    create_account_key = Column(String)
+    
+    def __repr__(self):
+        return "<User(uid=\"%s\", last_name=\"%s\", first_name=\"%s\")" %\
+            (self.uid, self.last_name, self.first_name)
+
+# set up argument parser
+def parse_args():
+    """
+    Set up the argument parser and parse all input arguments.
+    """
+    global verbose
+    parser = argparse.ArgumentParser()
+    parser.add_argument("filename",
+        help="registrar provided file of new students")
+    parser.add_argument("database", choices=["dev", "prod"],
+        help="select dev or prod database")
+    parser.add_argument("-v", "--verbose", help="increase output verbosity",
+        action="store_true")
+    args = parser.parse_args()
+    return args
+
+# parse the registrar provided file
+def parse_file(filename, verbose = False):
+    """
+    Parse the file and return an array of member objects, representing the new
+    members.
+    """
+    with open(filename) as registrar_file:
+        if verbose:
+            print "Loaded file \"" + args.filename + "\""
+        parsed_data = []
+        for raw_line in registrar_file:
+            # use regex to get all entries from between quotes
+            parsed_line = [None if entry == "" else entry
+                for entry in re.findall(r"\"(.*?)\"", raw_line)]
+            # instanciate a new member
+            new_member = Member(
+                last_name = parsed_line[1],
+                first_name = parsed_line[2],
+                middle_name = parsed_line[3],
+                uid = int(parsed_line[4]),
+                gender = 0 if parsed_line[5] == "M" else 1,
+                birthday = datetime.datetime
```

Comment:
I'm getting an error on parsing birthday.
It's because there's a non-standard format for months ("SEP").
See this legacy [script](https://github.com/ASCIT/donut-legacy/blob/master/utils/users/update_directory_data.py) for one way to approach it.